### DISCUSSION TITLE
Does virtink support mount a PVC into some path in vm?

### DISCUSSION THREAD

**Discussion by weixiao-huang** [smartxworks/virtink#39] [Q&A]
Sometimes, we want to mount a PVC into the VM to persistent some data. For example, we have a PVC called `my-data` and we want to mount it into `/data` path into vm. So does virtink support it?
To do this thing, I think there're two hands:
1. mount volume from host machine into VM (hostPath or local-path-provisioner)
2. mount volume from network (directly mount ceph rbd from network into VM)
So does virtink support one of the two method above?

**Comment by scuzhanglei** [comment]
yes, virtink already support PVC, 1) for Block PVC you can use it directly,  2) for filesystem PVC, need make sure a disk image file named as`disk.img`  in PVC root path. a better way is use [DataVolume](https://github.com/kubevirt/containerized-data-importer), which will create `disk.img` for filesystem PVC root path automatically. this is an [example](https://github.com/smartxworks/virtink/blob/main/samples/ubuntu-datavolume.yaml) VM with Datavolume.

**Comment by weixiao-huang** [comment]
But how to specify the volume's mount path in the VM? I've seen the docs about the [DataVolume](https://github.com/kubevirt/containerized-data-importer) but it seems that I didn't find some examples about specifying the volume mount path

**Comment by fengye87** [comment]
@weixiao-huang To mount a disk to a path inside a VM, you have several options:

- Use the `mount` command to mount the disk manually inside the VM
- If you want the mount to be persistent across reboots, you could add it to the `/etc/fstab` file
- You could also use cloud-init to do the mounting when booting the VM, please refer to cloud-init doc for how to do that

All Virtink could do is provide a volume as a VM disk to the VM, but the mounting part should be taken care of inside the VM.

**Comment by fengye87** [comment]
@weixiao-huang I'm closing this issue, feel free to reopen it if you want.
